rosidl_typesupport_opendds_c/resource/msg__type_support_c.cpp.em

```diff
@@ -471,15 +456,13 @@ _@(message.structure.namespaced_type.name)__to_cdr_stream(
     cdr_stream->allocator.deallocate(cdr_stream->buffer, cdr_stream->allocator.state);
     cdr_stream->buffer = static_cast<uint8_t *>(cdr_stream->allocator.allocate(cdr_stream->buffer_length, cdr_stream->allocator.state));
   }
-  // call the function again and fill the buffer this time
-  unsigned int buffer_length_uint = static_cast<unsigned int>(cdr_stream->buffer_length);
-  if (@(__dds_cpp_msg_type_prefix)_Plugin_serialize_to_cdr_buffer(
-      reinterpret_cast<char *>(cdr_stream->buffer),
-      &buffer_length_uint,
-      &dds_message) != true)
-  {
+  OpenDDS::DCPS::Message_Block_Ptr b(new ACE_Message_Block(size));
+  OpenDDS::DCPS::Serializer serializer(b.get(), false);
```

Comment:
So these Serializers are still using the native endianness regardless of what the other systems might be using. If there is no way to coordinate what endianness is being used I would suggest either always doing little endian:

```
  OpenDDS::DCPS::Serializer serializer(b.get(), !OpenDDS::DCPS::Serializer::SWAP_BE);
```

or always doing big endian:

```
  OpenDDS::DCPS::Serializer serializer(b.get(), OpenDDS::DCPS::Serializer::SWAP_BE);
```

Reply:
I should add most (if not basically all) systems use little endian, so it's probably more performant to use that if it's going to be fixed to one or the other.

Reply:
We will determine the requirement from external references, instead of making one up.

Reply:
If there is one, then yes I agree that would be the way to go. That's what I meant by "way to coordinate what endianness is being used".

Reply:
With regards to serialization and endianness, it should at least be compatible with the other RMW DDS implementations. 

While the process architectures are mostly little endian, I thought that the network byte order was typically big endian.

Reply:
> With regards to serialization and endianness, it should at least be compatible with the other RMW DDS implementations.

I was under the impression that wasn't necessarily a goal. If it is, then I can't say for certain what other RMWs are using, but can give a summary of what DDSes normally do. Normal DDS samples going over RTPS use an 8 byte alignment (`Serializer::ALIGN_CDR`, not left to default to `ALIGN_NONE` like it is here) and have a 4 byte header which says what the encoding and endianness of the stream is.

> While the process architectures are mostly little endian, I thought that the network byte order was typically big endian.

Normally I think there's something that says what endianness of the CDR is (like above) and so can it be written and read either way, and reordering the bytes as needed. I think OpenDDS encodes samples using the native endianness by default, but it can be changed using transport configuration.

Reply:
This is not really relevant to the ROSIDL Type Support.  As I was saying in the other comment, this is a bit a of strange case since it's not using IDL the way the DDS spec intends.

Anyway, there's a larger design discussion than we can have in comments of a PR, so let's leave all of that out of here.  It goes to requirements, not implementation.

Reply:
@kuznetsovmoci or @sonndinh, Please create a section in the RMW/ROSIDL document to define the endianess, byte alignment, etc. based on this conversation. We need to resolve how this should be done.